lib/sycamore/sycamore/writers/file_writer.py

```diff
@@ -176,3 +231,19 @@ def __init__(
     def write_row_to_file(self, row: dict[str, Any], file: NativeFile):
         binary = self._doc_to_bytes_fn(Document.from_row(row))
         file.write(binary)
+
+
+class _JsonBlockDataSink(BlockBasedFileDatasink):
+    def __init__(
+        self,
+        path: str,
+        filesystem: Optional[FileSystem] = None,
+    ) -> None:
+        super().__init__(path, filesystem=filesystem, filename_provider=BlockFilenameProvider())
+
+    def write_block_to_file(self, block: BlockAccessor, file: NativeFile) -> None:
+        for row in block.iter_rows(True):  # type: ignore[var-annotated]
+            doc = Document.from_row(row)
+            del doc.binary_representation  # Doesn't make sense in JSON
```

Comment:
Oh I see. So the comment above was just to warn other callers that might want to use document_to_bytes?

Reply:
Pretty much.  This particular bytes object was the only problem I ran into.  I have no idea what other values are possible.  In many cases, they may be UTF-8 strings.  It seems safer to drop this known field than to blanket-drop all bytes objects.  Time will tell.